### PR TITLE
feat: Add GET /aggregated-stats endpoint

### LINKED ISSUE #61: Aggregated data retrieval
Currently, the only way to get the aggregated stats is to get them directly from the database. A better option could be to expose some `/get_aggregated_stats` endpoint to avoid unnecessary coupling among services

### PR DESCRIPTION
**Pull request by sirewix**
Closes #61

### CODE REVIEW

File: config.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 database:
-  url: postgres://postgres:s3cr3t@localhost:5432/barad-dur
```
Comment (FlixCoder) [inline config.yaml:2]: Why did you remove the database name and change the secret and such?

File: Cargo.toml [hunk 2]
```diff
@@ -18,7 +18,7 @@ log = "0.4.17"
 once_cell = "1.16.0"
 serde = { version = "1.0.151", features = [ "derive" ] }
 serde_json = "1.0.91"
-sqlx = { version = "0.6.2", features = [ "runtime-tokio-rustls", "postgres", "macros", "migrate", "time", "json", "bigdecimal", "offline" ] }
```
Comment (FlixCoder) [inline Cargo.toml:21]: Was the offline feature removed or why is it gone?

File: src/server.rs [hunk 3]
```diff
@@ -3,19 +3,28 @@ use std::process;
...
+    tx: mpsc::Sender<model::Report>,
+) -> Result<()> {
     Server::bind(&settings.host.parse::<SocketAddr>()?)
         .serve(
             Router::new()
                 .route("/health", get(health_check))
                 .route("/report-usage-stats/push", put(save_report))
+                .route(
+                    "/aggregated-stats/:day",
+                    get(|day| get_aggregated_stats(db_settings, day)),
```
Comment (FlixCoder) [inline src/server.rs:26]: No need to manually add state to the functions. You can use axum state or axum extensions.
Also, the DB settings are just the connection settings, you most likely want to pass the DB pool already or it would connect every time again. Or how do the other endpoints do it?

File: Cargo.toml (hunk 2 above)
Comment (FlixCoder) [reply Cargo.toml:21]: ah I see, it is gone, so those sqlx files in the folder are the new way?

File: src/database.rs [hunk 4]
```diff
@@ -195,6 +195,20 @@ async fn aggregate_stats(pool: &sqlx::PgPool) -> Result<()> {
     Ok(())
 }
 
+pub async fn get_aggregated_stats(
+    db_settings: &DBSettings,
```
Comment (FlixCoder) [inline src/database.rs:199]:
```suggestion
    db_pool: &sqlx::PgPool,
```

File: config.yaml (hunk 1 above)
Comment (sirewix) [reply config.yaml:2]: Woops, the point was to remove `config.yaml` completely, I think it was added by accident at some point in the past since it's in `.gitignore`. Like everywhere we have only `config.sample.yaml` in repos, and not `config.yaml` themselves

File: Cargo.toml (hunk 2 above)
Comment (sirewix) [reply Cargo.toml:21]: Yeah, so I've spent couple of hours battling with a compilation error because I had newer version of `cargo sqlx`, which saved queries in `.sqlx` folder, but our code tried to read older `sqlx-data.json`. So we should either mention in readme that you need to install older version of sqlx or to just update our sqlx crate dependency (which I did here). 

And yeah, offline feature is removed, now it's default behavior, and instead of single `sqlx-data.json` file there's a `.sqlx` folder with each query in separate file

File: src/server.rs (hunk 3 above)
Comment (sirewix) [reply src/server.rs:26]: >  You can use axum state or axum extensions.

That's what I tried to do initially, but the rest of the code is written in such a way that the db access is isolated in `database.rs`. Also, functions in `database.rs` use [sorta like lazy static pattern](https://github.com/famedly/barad-dur/blob/25d36c2aeb2f7ae7a0df6d50e8fef011e56f13a0/src/database.rs#L73) which I was not sure I should touch. So that's why I do not pass pool to the endpoints handler. Other endpoints do not interact with db at all, only special threads and on of those threads is completely isolated (aggregate loop), the other is recieving data from channel. 

The other point why I abandoned state idea is because `save_report` (which uses `extract::Extension` would have a parameter (db settings or db pool) that it doesn't need, which would make [mocking it](https://github.com/famedly/barad-dur/blob/25d36c2aeb2f7ae7a0df6d50e8fef011e56f13a0/src/server.rs#L106) a bit problematic. It would be nice to solve it with something we recently discussed (`HasField` traits or smth), but currently rust doesn't have that. Another way would be to put some lenses when we construct a router, like:
```rust
  .route("/report-usage-stats/push", |s: Extension<State>, ...| put(save_report(s.only_things_related_to_save_report, ...))
```
But my first attempt resulted in some "temporary value created" or smth error, so I just did the easiest, but not prettiest thing.

Or we can start another db thread and use channels (or rather start tower service because we need request response here).

What do you think?

File: config.yaml (hunk 1 above)
Comment (emgrav) [reply config.yaml:2]: Makes sense, I think you should just delete config.yaml

File: src/server.rs (hunk 3 above)
Comment (emgrav) [reply src/server.rs:26]: If this works, I think it is good for now. We have plans to significantly refactor barad-dur at some point relatively soon as bumping f.x. axum will require some parts to be reworked. I think that would be a good opportunity to reevaluate the structure of the existing code as well, but let's go with this for now.

**Review by emgrav — CHANGES_REQUESTED** [review]
Let's just remove the accidentally commited config.yaml and then I think this is good to go

**Review by emgrav — APPROVED** [review]

**Comment by jcgruenhage** [comment]
For a service that is intentionally exposed directly to the internet without further authentication, having an unauthenticated endpoint that exposes data should *probably* be disabled by default to avoid others accidentally leaking data using that. `barad-dur` is an open source component also in use outside of Famedly

**Comment by sirewix** [comment]
> For a service that is intentionally exposed directly to the internet without further authentication

Why is it exposed to the internet? I don't think that is a good idea. We are exposing `PUSH` method to recieve statistics that directly modifies our database, that could easily be abused.
